lib/driver.js

```diff
@@ -54,6 +54,14 @@ module.exports = R.curry((server_url, Client) => {
       Client.put( DEFAULT_HEADERS, BucketUrl(bucket, docid), doc )
     )
 
+    // revid is the latest revision id needed to update an existing document
+  , upsert: R.curry((bucket, docid, revid, doc) => {
```

Comment:
So this PR can be closed/disregarded then? I will use the custom functions in our `kvasir/rfp.js` file in `node-api` in the meantime while you implement the methods